### DISCUSSION TITLE
TypeError: can't access property "name", nodeDef is undefined

### DISCUSSION THREAD

**Discussion by vortice3D** [google/model-viewer#4171] [Q&A]
Hi there everybody:

I have a GLB I'm generation from BabylonJS, using the export API.

I want to use that GLB into model-viewer, this web component raises the following exception:
`TypeError: can't access property "name", s is undefined`

The console warns about the following (space-opera.js 70437:25)

`TypeError: can't access property "name", nodeDef is undefined
    _loadNodeShallow https://modelviewer.dev/dist/space-opera.js:68848
    loadNode https://modelviewer.dev/dist/space-opera.js:68777
    dependency https://modelviewer.dev/dist/space-opera.js:67436
    _invokeOne https://modelviewer.dev/dist/space-opera.js:67385
    getDependency https://modelviewer.dev/dist/space-opera.js:67434
    loadAnimation https://modelviewer.dev/dist/space-opera.js:68585
    dependency https://modelviewer.dev/dist/space-opera.js:67488
    _invokeOne https://modelviewer.dev/dist/space-opera.js:67385
    getDependency https://modelviewer.dev/dist/space-opera.js:67486
    dependencies https://modelviewer.dev/dist/space-opera.js:67538
    getDependencies https://modelviewer.dev/dist/space-opera.js:67536
    parse https://modelviewer.dev/dist/space-opera.js:67232
`

I'm pretty sure that is a problem with the ThreeJS import GLB API, but it is opened without problems with BabylonJS Sandbox, Windows 10 3D Viewer, Blender, ...

So how can I debug what is happening here?

Thanks for your time!

**Comment by elalish** [comment]
Is it valid? https://github.khronos.org/glTF-Validator/

Often some importers handle certain types of invalid files, but it can't be relied upon. If it has no errors, then file a bug on the three.js GLTFImporter.

**Reply by vortice3D** [reply]
Hi, @elalish:

My GLTF has some "hints" from origin, that is from the Maya exporter, but is difficult to solve all of them. Anyway, I think the ThreeJS GLTFImporter is complaining here about some specific nodes, related to animated instances.

How can I get any information about the problematic nodes (i.e. its names) within the importer?

Thanks for your time!

**Reply by elalish** [reply]
You can use Chrome dev tools to break on exception and then look through the call stack. Can you try it here? https://threejs.org/editor/ If it also fails to load there, then you should file an issue on three.js and include your model.

**Reply by vortice3D** [reply]
Thanks for the info, @elalish:

As you know, usually the models (GLTF/GLB) have errors, but however they work well on WebGL. For example, even in the [model-viewer editor](https://modelviewer.dev/editor/) you have the robot model that is failing in the validation check, with errors.

The model I was talking you about, do not load in the ThreeJS editor (I've checked it in the first time), and indeed the exception I mentioned is also there the cause for doing so.

Following your advice, I'll check the call stack in order to try to identify the problematic node(s) in my file.
